File: backend/src/Financas.Api/Domain/Models/Titulo.cs
using System.ComponentModel.DataAnnotations;
using Financas.Api.Domain.Enums;

namespace Financas.Api.Domain.Models
{
    public class Titulo
    {
        [Key]
        public long Id { get; set; }

        [Required]
        public TipoTituloEnum? TipoTitulo { get; set; }

        [Required]
        [Range(1, long.MaxValue, ErrorMessage = "O ID do usuário deve ser maior que zero.")]
        public long IdUsuario { get; set; }

        public Usuario Usuario { get; set; }

        [Required]
        [Range(1, long.MaxValue, ErrorMessage = "O ID da natureza de lançamento deve ser maior que zero.")]
        public long IdNaturezaDeLancamento { get; set; }
        public NaturezaDeLancamento NaturezaDeLancamento { get; set; }

        [Required(ErrorMessage = "O campo de descrição é obrigatório.")]
        public string Descricao { get; set; } = string.Empty;

        [Required]
        [Range(0.01, double.MaxValue, ErrorMessage = "O valor original deve ser maior que zero.")]
        public double ValorOriginal { get; set; }

        public string Observacao { get; set; } = string.Empty;

        [Required]
        public DateTime DataCadastro { get; set; }

        public DateTime? DataPagamento { get; set; }
        public DateTime? DataInativacao { get; set; }
    }
}

File: frontend/financas/src/models/Titulo.js
import conversorData from "@/utils/conversorData";

export default class Titulo {
  constructor(obj = {}) {
    this.id = obj.id;
    this.idUsuario = obj.idUsuario;
    this.idNaturezaDeLancamento = obj.idNaturezaDeLancamento;
    this.naturezaDeLancamentoDescricao = obj.naturezaDeLancamentoDescricao;
    this.descricao = obj.descricao;
    
    this.tipoTitulo = obj.tipoTitulo !== undefined && obj.tipoTitulo !== null ? obj.tipoTitulo : null; 
    
    this.dataCadastro = conversorData.aplicarMascaraEmDataIso(obj.dataCadastro);
    this.dataInativacao = obj.dataInativacao;
    this.observacao = obj.observacao;
    this.valorOriginal = obj.valorOriginal;
    this.dataPagamento = obj.dataPagamento;
  }

  static tipoTituloMap = {
    0: "Gasto",
    1: "Recebimento",
  };

  get tipoTituloDescricao() {
    return Titulo.tipoTituloMap[this.tipoTitulo] || "Desconhecido";
  }

  modeloValidoParaCadastro() {
    return !!(this.descricao && this.naturezaDeLancamentoDescricao && this.valorOriginal && this.dataPagamento && this.tipoTitulo !== undefined);
  }

  modeloValidoParaAtualizar() {
    return !!(this.descricao && this.naturezaDeLancamentoDescricao && this.valorOriginal && this.dataPagamento && this.tipoTitulo !== undefined);
  }
}
